sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -21,6 +22,25 @@ public class HttpHeaders implements Iterable<HttpHeader> {
     // This map is a case-insensitive key (i.e. lower-cased), but the returned HttpHeader key will be as-provided to us
     private final Map<String, HttpHeader> headers;
 
+    static {
+        HttpHeadersHelper.setAccessor(new HttpHeadersHelper.HttpHeadersAccessor() {
+            @Override
+            public HttpHeaders setNoKeyFormat(HttpHeaders headers, String formattedName, String name, String value) {
+                return headers.setInternal(formattedName, name, value);
+            }
+
+            @Override
+            public HttpHeader getNoKeyFormat(HttpHeaders headers, String formattedName) {
+                return headers.getInternal(formattedName);
+            }
+
+            @Override
+            public String getValueNoKeyFormat(HttpHeaders headers, String formattedName) {
+                return headers.getValueInternal(formattedName);
+            }
```

Comment:
I wonder if we should use a pattern of `<oldName>_<modifier>`, e.g. `set_noKeyFormatting`, `get_noKeyFormatting`, and `getValue_noKeyFormatting`